utils/nft-utils/src/main.rs

```diff
@@ -0,0 +1,217 @@
+        },
+    );
+    Transaction::<DefaultContext>::new_signed_tx(
+        signer,
+        create_collection_message.try_to_vec().unwrap(),
+        nonce,
+    )
+}
+
+async fn mint_nft(
+    signer: &DefaultPrivateKey,
+    nonce: u64,
+    collection_name: &str,
+    token_id: u64,
+    owner: &Address,
+) -> Transaction<DefaultContext> {
+    let collection_address = get_collection_address::<DefaultContext>(
+        collection_name,
+        signer.default_address().as_ref(),
+    );
+    let token_uri = get_nft_metadata_url(&collection_address.to_string(), token_id);
+    let mint_nft_message =
+        RuntimeCall::<DefaultContext, MockDaSpec>::nft(CallMessage::<DefaultContext>::MintNft {
+            collection_name: collection_name.to_string(),
+            token_uri,
+            token_id,
+            owner: UserAddress::new(owner),
+            frozen: false,
+        });
+    Transaction::<DefaultContext>::new_signed_tx(
+        signer,
+        mint_nft_message.try_to_vec().unwrap(),
+        nonce,
+    )
+}
+
+async fn transfer_nft(
+    signer: &DefaultPrivateKey,
+    nonce: u64,
+    collection_address: &CollectionAddress<DefaultContext>,
+    token_id: u64,
+    to: &Address,
+) -> Transaction<DefaultContext> {
+    let transfer_message = RuntimeCall::<DefaultContext, MockDaSpec>::nft(CallMessage::<
+        DefaultContext,
+    >::TransferNft {
+        collection_address: collection_address.clone(),
+        token_id,
+        to: UserAddress::new(to),
+    });
+    Transaction::<DefaultContext>::new_signed_tx(
+        signer,
+        transfer_message.try_to_vec().unwrap(),
+        nonce,
+    )
+}
+
+#[tokio::main]
+async fn main() {
+    let creator_pk = DefaultPrivateKey::try_from(&PK1[..]).unwrap();
+    let owner_1_pk = DefaultPrivateKey::try_from(&PK2[..]).unwrap();
+    let owner_2_pk = DefaultPrivateKey::try_from(&PK3[..]).unwrap();
+
+    let client = SimpleClient::new("localhost", 12345).await.unwrap();
+
+    let mut nonce = 0;
+    let mut nft_id = 1;
+    let mut transactions = vec![];
+    transactions.push(create_collection(&creator_pk, nonce, COLLECTION_1).await);
+    nonce += 1;
+    transactions.push(create_collection(&creator_pk, nonce, COLLECTION_2).await);
+    nonce += 1;
+    transactions.push(create_collection(&creator_pk, nonce, COLLECTION_3).await);
+
+    client.send_transactions(transactions, None).await.unwrap();
+    // sleep is necessary because of how the sequencer currently works
+    // without the sleep, there is a concurrency issue and some transactions would be ignored
+    thread::sleep(Duration::from_millis(1000));
+
+    let mut transactions = vec![];
+    for _ in 0..15 {
+        nonce += 1;
+        transactions.push(
+            mint_nft(
+                &creator_pk,
+                nonce,
+                COLLECTION_1,
+                nft_id,
+                &owner_1_pk.default_address(),
+            )
+            .await,
+        );
+        nft_id += 1;
+    }
```

Comment:
nit: We have several of these loops with mutable nonces etc. it's not the end of the world, but feels a bit messy. Maybe we can simplify with an iterator, and/or make a helper function?